brainprep/qc_utils.py

```diff
@@ -0,0 +1,931 @@
+        the mean correlation data.
+    output_pdf: filename of the output pdf.
+    limit : maximal number of diapo in the pdf
+        default=None
+
+    Saving
+    -------
+    Nii images Diaporama sorted by mean correlation saved in pdf format.
+    columns: wm cat12vbm NII image (1 slice), mwp1 NII images (6 slices)
+    rows:sagittal, coronal, axial mri.
+
+
+    """
+    # initialise size
+    max_range = limit or len(nii_filenames)
+    # create pdf output
+    pdf = PdfPages(output_pdf)
+    # plot slices
+    for i, nii_file in list(enumerate(nii_filenames))[:max_range]:
+        fig = plt.figure(figsize=(30, 20))
+        gs = GridSpec(3, 7, figure=fig)
+        # t1w ori
+        ax1 = fig.add_subplot(gs[0, 0])
+        ax2 = fig.add_subplot(gs[1, 0])
+        ax3 = fig.add_subplot(gs[2, 0])
+        # qr
+        ax4 = fig.add_subplot(gs[0, 1:])
+        ax5 = fig.add_subplot(gs[1, 1:])
+        ax6 = fig.add_subplot(gs[2, 1:])
+        nii = nibabel.load(nii_file)
+        nii_ref = nii_file.replace("_desc-6apply", "")
+        nii_ref = nii_ref.replace("/derivatives/quasi-raw/", "/rawdata/")
+        if os.path.exists(nii_ref):
+            pass
+        else:
+            nii_ref = nii_ref.replace("/psy_sbox/", "/psy/")
+        assert os.path.exists(nii_ref), nii_ref
+        nii_ref = nibabel.load(nii_ref)
+        plt.suptitle('Subject %s, session %s, run %s with mean '
+                     'correlation %.3f'
+                     % (mean_corr[i][0], mean_corr[i][1],
+                        mean_corr[i][2], mean_corr[i][3]),
+                     fontsize=40)
+        # t1w ori
+        plotting.plot_anat(nii_ref, figure=fig, axes=ax1, dim=0, cut_coords=1,
+                           display_mode='x')
+        plotting.plot_anat(nii_ref, figure=fig, axes=ax2, dim=0, cut_coords=1,
+                           display_mode='y')
+        plotting.plot_anat(nii_ref, figure=fig, axes=ax3, dim=0, cut_coords=1,
+                           display_mode='z')
+        # qr
+        plotting.plot_anat(nii, figure=fig, axes=ax4, dim=1,
+                           cut_coords=6, display_mode='x')
+        plotting.plot_anat(nii, figure=fig, axes=ax5, dim=1,
+                           cut_coords=6, display_mode='y')
+        plotting.plot_anat(nii, figure=fig, axes=ax6, dim=1,
+                           cut_coords=6, display_mode='z')
+        # resize
+        plt.subplots_adjust(wspace=0, hspace=0, top=0.9, bottom=0.1)
+        # save
+        pdf.savefig()
+        plt.close(fig)
+    pdf.close()
+
+
+def pdf_cat(pdf_filenames, output_pdf):
+    """Concatenation of pdf files in one big pdf.
+
+    Parameters
+    ----------
+    pdf_filenames: list
+        filenames of input data.
+    output_pdf: filename of the output pdf.
+
+    Saving
+    -------
+    Big cat12vbm pdf reports sorted by mean correlation.
+
+
+    """
+    pdfWriter = PdfFileWriter()
+    for file in pdf_filenames:
+        pdfFileObj = open(file, 'rb')
```

Comment:
It is recommended to use only lower case in variable names.

```suggestion
        pdf_file_obj = open(file, 'rb')
```

Seems not closed. Prefer the with statement synthax:

```
with open(file, 'rb') as pdf_file_obj:
     some code
```